Automate to Automate HA Migration docs (#7810)
chef/automate: components/docs-chef-io/content/automate/ha_automate_to_automate_ha.md

{{< warning >}}

- Standalone Chef Automate or Chef Automate with embedded Chef Infra Server can migrate to Automate HA, with the minimum version of Chef Automate: [2020xxxxxxxx](https://docs.chef.io/release_notes_automate/)

{{< /warning >}}

## Migration with FileSystem Backup Locally

Follow the steps below when migrating to On-Premises or AWS HA deployment (but not for AWS with managed services).

1. Create a Backup of Chef Automate Standalone using the following command:

    1. Run the below command to create the backup in the `/var/opt/chef-automate/backups` location unless you specify the location in the `config.toml` file.

        ```bash
        chef-automate backup create
        ```

        Once the backup is completed, save the backup Id. For example: `20210622065515`.

    1. Run the below command to create the `bootstrap.abb` bundle. This bundle captures any local credentials or secrets not persisted in the database.

        ```bash
        chef-automate bootstrap bundle create bootstrap.abb
        ```

> 1. If you haven't specified the location in the `config.toml` file, go to the `/var/opt/chef-automate/backups` location and create **Bundle** using the following command:

Reviewer: ```suggestion
1. Go to the backup location mentioned in Automate config, but if you haven't specified the location go to the `/var/opt/chef-automate/backups` location and create **Bundle** using the following command:
```

    ```bash
    tar -cvf backup.tar.gz <backup_id>/ automatebackup-elasticsearch/ .tmp/
    ```

> 1. Transfer the `tar` bundle to one of the Chef-Automate nodes of Automate HA using the following command:

Reviewer: ```suggestion
1. Transfer the `tar` bundle to one of the Chef Automate nodes of Automate HA using the following command:
```

    ```bash

>         scp -i /path/to/key /path/to/backup-file user@host:/home/user

Reviewer: ```suggestion
        scp -i </path/to/key> </path/to/backup-file> <user>@<host>:/home/<user>
```

    ```

1. Transfer the `bootstrap.abb` file to all the Chef Automate HA FrontEnd Nodes (both Chef Automate and Chef Infra Server) using the following command:

    ```bash

>         scp -i /path/to/key /path/to/bootstrap.abb user@host:/home/user

Reviewer: ```suggestion
        scp -i </path/to/key> </path/to/bootstrap.abb> <user>@<host>:/home/<user>
```

    ```

1. Go to Bastion and:
    - Create a `.toml` (say os_config.toml) file in the Bastion host. Once done, copy the following contents to the `.toml` file and patch the file in all the OpenSearch nodes.

    ```bash
    [path]
      repo = "/mnt/automate_backups"
    ```

    The following command will patch the configuration in all the OpenSearch nodes.

    ```bash
    chef-automate config patch --opensearch <path to os_config.toml>
    ```

    - Create a `.toml` (say automate.toml) file in the Bastion host. Once done, copy the following content to the `.toml` file and then patch the file in all the Frontend nodes.

    ```bash
    [global.v1.external.opensearch.backup]
        enable = true
        location = "fs"
    [global.v1.external.opensearch.backup.fs]
        path = "/mnt/automate_backups"
    [global.v1.backups.filesystem]
        path = "/mnt/automate_backups"
    ```

    The following command will patch the configuration in all the Frontend nodes:

    ```bash
    chef-automate config patch --fe <Path to automate.toml>
    ```

> 1. Go to the Chef-Automate node of Automate HA cluster, where we copied the `tar` file. Unzip the bundle using the following:

Reviewer: ```suggestion
1. Go to the Chef Automate node of Automate HA cluster, where we copied the `tar` file. Unzip the bundle using the following:
```

    ```bash
    tar -xf backup.tar.gz -C /mnt/automate_backups
    ```

> 1. Run the following command at the Chef-Automate node of Automate HA cluster to get the applied `config`:

Reviewer: ```suggestion
1. Run the following command at the Chef Automate node of Automate HA cluster to get the applied `config`:
```

    ```bash
    [global.v1.external.opensearch.backup]
        enable = true
        location = "fs"
    [global.v1.external.opensearch.backup.fs]
        path = "/mnt/automate_backups"
    [global.v1.backups.filesystem]
        path = "/mnt/automate_backups"
    ```

    The following command will patch the configuration in all the Frontend nodes:

    ```bash
    chef-automate config patch --fe <Path to automate.toml>
    ```

1. Create a Backup of Chef Automate Standalone using the following command:

    1. Run the below command to create the backup in the `/var/opt/chef-automate/backups` location unless you specify the location in the `config.toml` file.

        ```bash
        chef-automate backup create
        ```

        Once the backup is completed, save the backup Id. For example: `20210622065515`.

    1. Run the below command to create the `bootstrap.abb` bundle. This bundle captures any local credentials or secrets not persisted in the database.

        ```bash
        chef-automate bootstrap bundle create bootstrap.abb
        ```

> 1. Run the following command at the Chef-Automate node of Automate HA cluster to get the applied config:

Reviewer: ```suggestion
1. Run the following command on the Chef Automate node of Automate HA cluster to get the applied config:
```

{{< warning >}}

> - The Minimum version of Automate to perform migration is 4.x.

Reviewer 2: please mark the min version as 2020xxxxxxxx

- Customers in 2022xxxx or 3.x version needs to update to 4.x version.
- Automate HA with AWS managed needs to use OpenSearch version <1.3.2
- The OpenSearch version in Automate should be the same version as Automate HA.

{{< /warning >}}

> ## Upgrade with FileSystem Backup Locally

Reviewer 2: update to `Migration`

> Follow the below steps for Automate HA when you are upgrading with On-Premises and AWS deployment (but not for AWS managed services).

Reviewer 2: This statement is not meaningful

1. Create a Backup of Chef Automate Standalone using the following command:

    Run the below command to create the backup in the `/var/opt/chef-automate/backups` location unless you specify the location in the `config.toml` file.

    ```bash
    chef-automate backup create
    ```

    Run the below command command to create the `bootstrap.abb` bundle. This bundle captures any local credentials or secrets not persisted in the database.

    ```bash
    chef-automate bootstrap bundle create bootstrap.abb
    ```

    Once the backup is completed, save the backup Id. For example: `20210622065515`.

1. If you haven't specified the location in the `config.toml` file, go to the `/var/opt/chef-automate/backups` location and create **Bundle** using the following command:

    ```bash
    tar -cvf backup.tar.gz <backup_id>/ automatebackup-elasticsearch/ .tmp/
    ```

> 1. Transfer the `tar` bundle to one of the Chef Automate HA Frontend Nodes using the following command:

Reviewer 2: We have to transfer the file to Automate node, not the fronted node. Infact we have to do the restoration in Bastion node.

    {{% automate/char-warn %}}
    {{< /warning >}}

1. Transfer the `bootstrap.abb` file to all the Chef Automate HA FrontEnd Nodes (both Chef Automate and Chef Infra Server) using the following command:

    ```bash

>         scp -i /path/to/key /path/to/bootstrap.abb user@host:/home/user

Reviewer: ```suggestion
        scp -i </path/to/key> </path/to/bootstrap.abb> <user>@<host>:/home/<user>
```

    ```

1. Unpack the `bootstrap.abb` file on all the Frontend nodes:

    Login to Each Frontend Node and then run after copying the `bootstrap.abb` file.

    ```bash
        chef-automate bootstrap bundle unpack bootstrap.abb
    ```

> 1. Stop all the frontend instances except where you saved the `current_config.toml` file at automate nodes in Automate HA Cluster. Run the following command to all the Automate and Chef Infra Server nodes:

Reviewer: ```suggestion
1. Stop all the frontend instances except where you saved the `current_config.toml` file on Chef Automate nodes in Automate HA Cluster. Run the following command to all the Automate and Chef Infra Server nodes:
```

    # endpoint (required): The endpoint for the region the bucket lives in for Automate Version 3.x.y
    # endpoint (required): For Automate Version 4.x.y, use this https://s3.amazonaws.com
    endpoint = "https://s3.amazonaws.com"
    # base_path (optional):  The path within the bucket where backups should be stored
    # If base_path is not set, backups will be stored at the root of the bucket.
    base_path = ""
    [global.v1.backups.s3.credentials]
    # Optionally, AWS credentials may be provided. If these are not provided, the IAM instance
    # credentials will be used. It's also possible for these to be read through the standard
    # AWS environment variables or through the shared AWS config files.
    access_key = "<access_key>"
    secret_key = "<secret_key>"
    session_key = "<session_key>"
    [global.v1.backups.s3.ssl]
    # root_cert (optional): The root certificate used for SSL validation.
    # For S3 compatible APIs, you can set the SSL root cert if needed
    root_cert = """
    -----BEGIN CERTIFICATE-----
    ...
    -----END CERTIFICATE-----
    """
    ```

1. Create a Backup of Chef Automate Standalone using the following command:

>     1. Run the below command to create the backup in the `/var/opt/chef-automate/backups` location unless you specify the location in the `config.toml` file.

Reviewer: ```suggestion
    1. Run the below command to create the backup in the `/var/opt/chef-automate/backups` location unless you specify the location in the Standalone Automate config.
```

    {{% automate/char-warn %}}
    {{< /warning >}}

1. On every Frontend Node, copy the `bootstrap.abb` file and unpack the file using the following command:

    ```bash
    chef-automate bootstrap bundle unpack bootstrap.abb
    ```

> 1. Stop all the frontend instances except where you saved the `current_config.toml` file at automate node in Automate HA Cluster. Run the following command to all the Automate and Chef Infra Server nodes:

Reviewer: ```suggestion
1. Stop all the frontend instances except where you saved the `current_config.toml` file on Chef Automate node in Automate HA Cluster. Run the following command to all the Automate and Chef Infra Server nodes:
```
